### PR TITLE
add fb_reprepro cookbook

### PR DESCRIPTION
**Pull request by davide125**
This adds a cookbook to setup APT repositories using reprepro. It's a rework of an unpublished cookbook I've been running at home for a while. Tested on Debian.

### CODE REVIEW

**Comment by facebook-github-bot-9** [comment]
@davide125 updated the pull request - [view changes](https://github.com/facebook/chef-cookbooks/pull/20/files/80fab1a762cffe9603655aa5192a7998409b95ae..707991c625a43eb4ef3610b3fe9ff64b5fbdec7e)

File: cookbooks/fb_reprepro/recipes/default.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  fail 'fb_reprepro is only supported on Debian and Ubuntu.'
+end
+
+package 'reprepro' do
+  action :upgrade
+end
+
+directory 'repository' do
+  only_if { node['fb_reprepro']['options']['basedir'] }
+  name lazy { node['fb_reprepro']['options']['basedir'] }
```
Comment (jaymzh) [inline cookbooks/fb_reprepro/recipes/default.rb]: I believe it's `path`

File: cookbooks/fb_reprepro/recipes/default.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  owner lazy { node['fb_reprepro']['user'] }
+  group lazy { node['fb_reprepro']['group'] }
+  mode '0644'
+end
+
+directory 'repository/conf' do
+  only_if { node['fb_reprepro']['options']['basedir'] }
+  name lazy { "#{node['fb_reprepro']['options']['basedir']}/conf" }
```
Comment (jaymzh) [inline cookbooks/fb_reprepro/recipes/default.rb]: same

**Comment by facebook-github-bot-0** [comment]
@davide125 updated the pull request - [view changes](https://github.com/facebook/chef-cookbooks/pull/20/files/7f96f43da90b2c1789ce4280f317829ef9a00343..0078a24f5b24ddb817560be3b39af9ac95b7ad21)

**Comment by davide125** [comment]
Updated. The rubocop failures are due to https://github.com/facebook/chef-cookbooks/pull/26#issuecomment-248190513

**Comment by facebook-github-bot-0** [comment]
@davide125 updated the pull request - [view changes](https://github.com/facebook/chef-cookbooks/pull/20/files/0078a24f5b24ddb817560be3b39af9ac95b7ad21..fa3e04e4ea7b8dfa9721d416d53460c228fe1b2c) - [changes since last import](https://github.com/facebook/chef-cookbooks/pull/20/files/0078a24f5b24ddb817560be3b39af9ac95b7ad21..fa3e04e4ea7b8dfa9721d416d53460c228fe1b2c)

**Pull request closed without merging**